deploy-service/common/src/main/java/com/pinterest/deployservice/events/EventBridgePublisher.java

```diff
@@ -53,6 +55,9 @@ public void publish(BuildBean buildBean, String action) {
     } catch (Exception e) {
       logger.error("Failed to publish event to Event Bridge: {}", entry, e);
     }
+
+    // set branch name back to its original value in case it's expected.
+    buildBean.setScm_branch(originalBranch);
```

Comment:
Best practice would generally be to make a copy instead of modifying the input object. In the case that multiple threads are referencing the object there could be a race condition. And even if it doesn't matter today maybe it could in the future.

Not blocking though.